plugin/helpers_single_node.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package plugin
+
+import "errors"
+
+// RunOnSingleNode is a wrapper function which should guarantee that only one plugin instance can run a function associated with the given id at a time
+// The id parameter is an identifier that uses for synchronization must be unique between each function.
+func (p *HelpersImpl) RunOnSingleNode(id string, f func()) (bool, error) {
+	updated, err := p.KVCompareAndSetJSON(id, nil, true)
```

Comment:
We should prefix the key with something like `runOnSignleNodeLock`(See https://github.com/mattermost/mattermost-server/pull/12969/files)